src/sqlalchemy_cratedb/dialect.py

```diff
@@ -228,8 +228,12 @@ def connect(self, host=None, port=None, *args, **kwargs):
         servers = to_list(server)
         if servers:
             use_ssl = asbool(kwargs.pop("ssl", False))
-            if use_ssl:
+            # TODO: Switch to the canonical default `sslmode=prefer` later.
+            sslmode = kwargs.pop("sslmode", "disable")
+            if use_ssl or sslmode in ["allow", "prefer", "require", "verify-ca", "verify-full"]:
                 servers = ["https://" + server for server in servers]
+            if sslmode == "require":
+                kwargs["verify_ssl_cert"] = False
             return self.dbapi.connect(servers=servers, **kwargs)
         return self.dbapi.connect(**kwargs)
```

Comment:
When evaluating the patch downstream using `sqlalchemy-cratedb==0.42.0.dev2`, we found that this connection string syntax, omitting the host name, is not supported yet: `crate://?sslmode=require`.

Currently, it bails out like this:
```python
TypeError: Connection.__init__() got an unexpected keyword argument 'sslmode'
```

The reason is, as you can observe above, that two different code paths are used here.